Legacy/bonej/src/main/java/org/bonej/plugins/ParticleAnalysis.java

```diff
@@ -516,6 +520,45 @@ public void filterParticles(final ImagePlus imp, final int[][] particleLabels, f
 		}
 		return eulerCharacters;
 	}
+	
+	/**
+	 * Calculate number of branches and total branch length by running 
+	 * Skeletonize3D and Analyze Skeleton on each particle.
+	 * 
+	 * Uses AnalyzeSkeleton's run() method with no arguments.
+	 * 
+	 * @param imp Input image
+	 * @param particleLabels particle label array
+	 * @param limits xyz limits of each particle
+	 * @param nParticles number of particles
+	 * @return array of AnalyzeSkelton result objects.
+	 * 	Use methods therein to get specific results out. Each result is a list of trees,
+	 * but there should be only one tree per particle so its result is at [0].
+	 */
+	static SkeletonResult[] getBranchLength(final ImagePlus imp,
+			final int[][] particleLabels, final int[][] limits, final int nParticles)
+	{
+		final SkeletonResult[] skeletonResults = new SkeletonResult[nParticles];
+		Thread[] threads = Multithreader.newThreads();
+		AtomicInteger ai = new AtomicInteger(1);
+		for (int thread = 0; thread < threads.length; thread++) {
```

Comment:
Are `AnalyzeSkeleton_` and `Skeletonize` thread safe?

Reply:
No idea, will test.

Reply:
(nothing spewed out on me so far, but I will try again with more faster threads)